### PR TITLE
Collapse assertions after conversion

### PR DESCRIPTION
**Pull request by timtebeek**
We found that sometimes a second pass was necessary before converted assertions were collapsed. This PR improves the current state.

### CODE REVIEW

File: src/test/java/org/openrewrite/java/testing/hamcrest/HamcrestMatcherToAssertJTest.java [hunk 1]
```diff
@@ -385,7 +386,8 @@ void test() {
         @Test
         void containsInAnyOrderWithArray() {
             rewriteRun(
-              spec -> spec.recipe(new HamcrestMatcherToAssertJ("containsInAnyOrder", "containsExactlyInAnyOrder", null)),
+              spec -> spec.recipe(new HamcrestMatcherToAssertJ("containsInAnyOrder", "containsExactlyInAnyOrder", null))
+                .afterTypeValidationOptions(TypeValidation.all().methodInvocations(false)),
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/testing/hamcrest/HamcrestMatcherToAssertJTest.java:390]: This was the only one failing with the new way of composing the template and arguments. For now could not figure out why within the timebox, but figured it's a good enough improvement to tolerate this edge case imperfection.

File: src/test/java/org/openrewrite/java/testing/assertj/AssertJBestPracticesTest.java [hunk 2]
```diff
@@ -101,6 +101,124 @@ void testMethod(List<String> list) {
...
+                  """,
+                """
+                  import java.util.List;
+
+                  import static org.assertj.core.api.Assertions.assertThat;
+
+                  class BiscuitTest {
+                      void biscuits() {
+                          List<String> biscuits = List.of("Ginger", "Chocolate", "Oatmeal");
+                          assertThat(biscuits).isNotNull();
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/testing/assertj/AssertJBestPracticesTest.java:135]: Ideally this one line would be removed as well, but it's already an improvement to see the others collapsed, where before we saw none collapsed until a second execution.

File: src/main/java/org/openrewrite/java/testing/hamcrest/HamcrestMatcherToAssertJ.java [hunk 3]
```diff
@@ -154,9 +154,10 @@ private String typeToIndicator(JavaType type) {
                         type.toString().replaceAll("<.*>", "") : "java.lang.Object";
                 return String.format("#{anyArray(%s)}", str);
             }
-            String str = type instanceof JavaType.Primitive || type.toString().startsWith("java.") ?
-                    type.toString().replaceAll("<.*>", "") : "java.lang.Object";
-            return String.format("#{any(%s)}", str);
+            if (type instanceof JavaType.Primitive || type.toString().startsWith("java.")) {
+                return "#{any()}";
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/testing/hamcrest/HamcrestMatcherToAssertJ.java:158]: In these cases we use the original type from the argument, as opposed to a simplified type we used before.
